ui/pages/Login/components/Login.jsx

```diff
@@ -12,6 +16,48 @@ const FIELDS = [
   { name: 'password', label: 'Password', type: 'password', validate: validators.required },
 ]
 
+const googleLogin = () => {
+  return new HttpRequestHelper('/api/login_google',
+    (responseJson) => {
+      // Redirect to google auth website
+      const width = 600
+      const height = 800
+      const left = (window.screen.width - width) / 2
+      const top = (window.screen.height - height) / 2
+      const params = `scrollbars=no,status=no,location=no,toolbar=no,menubar=no,
+        width=${width},height=${height},left=${left},top=${top}`
+      const win = window.open(responseJson.data, 'Google Sign In', params)
+      win.focus()
+    },
+    (e) => {
+      throw new SubmissionError({ _error: [e.message] })
+    },
+  ).get()
+}
+
+export const OAuth2Callback = ({ location }) => {
+  // Send the authentication results to the backend to finish the logging-in procedure
```

Comment:
this is really bad practice for dispatching events in react, you don't just call functions in a raw render method. Maybe just use the `DataLoader` component with the login function as the `load` prop and `loading` always set to true (or even better, actually set the loading and error props properly when you dispatch the login event)

Reply:
Okay. I use to call these kinds of event dispatching after component mounted but I can't find an easy way to do here. I'm changing it to use the `DataLoader`.

Reply:
Changed.